src/provider/provider_os_memory.c

```diff
@@ -337,7 +337,8 @@ static umf_result_t os_free(void *provider, void *ptr, size_t size) {
 
     errno = 0;
     int ret = os_munmap(ptr, size);
```

Comment:
why we can't check size inside os_munmap?

Reply:
Because the behavior must be the same for all implementations of all OSes.